apis/hive/v1/hiveconfig_types.go

```diff
@@ -88,9 +88,72 @@ type HiveConfigSpec struct {
 	// +optional
 	ControllersConfig *ControllersConfig `json:"controllersConfig,omitempty"`
 
+	// AWSPrivateLink defines the configuration for the aws-private-link controller.
+	// It provides 3 major information required by the controller,
+	// 1. The Credentials that should be used to create AWS PrivateLink resources other than
+	//     what exist in the customer's account.
+	// 2. A list of VPCs that can be used by the controller to choose one to create AWS VPC Endpoints
+	//     for the AWS VPC Endpoint Services created for ClusterDeployments in their
+	//     corresponding regions.
+	// 3. A list of VPCs that should be able to resolve the DNS addresses setup for Private Link.
+	AWSPrivateLink *AWSPrivateLinkConfig `json:"awsPrivateLink,omitempty"`
+
 	FeatureGates *FeatureGateSelection `json:"featureGates,omitempty"`
 }
 
+// AWSPrivateLinkConfig defines the configuration for the aws-private-link controller.
+type AWSPrivateLinkConfig struct {
+	// CredentialsSecretRef references a secret in the TargetNamespace that will be used to authenticate with
+	// AWS for creating the resources for AWS PrivateLink.
+	CredentialsSecretRef corev1.LocalObjectReference `json:"credentialsSecretRef"`
+
+	// VPCEndpointInventory is a list of VPCs and the corresponding subnets in various AWS regions.
+	// The controller uses this list to choose a VPC for creating AWS VPC Endpoints. Since the
+	// VPC Endpoints must be in the same region as the ClusterDeployment, we must have VPCs in that
+	// region to be able to setup Private Link.
+	VPCEndpointInventory []AWSPrivateLinkInventory `json:"vpcEndpointInventory,omitempty"`
```

Comment:
So here we have vpcEndpointInventory but below we have associatedVPCs (inverts the position of VPC). Both look like a list of VPC IDs and regions, one with additional subnet array.

One is using VPCID, the other just ID.

Was the AWSPrivateLinkVPC meant to be embedded in AWSPrivateLinkInventory as well? Right now it's separated but only used in AWSAssociatedVPC.

Feels like these are very similar and should look quite similar in the API.

endpointVPCInventory
- region
  vpcID
  subnets
associatedVPCs:
- region
  vpcID

Reply:
@dgoodwin  fixed the api for hiveconfig in f3d3954 

new one looks like:
```yaml
endpointsVPCInventory:
- region
  vpcID:
  subnets:
  - availabilityZone:
    subnetID:
associatedVPCs:
- region:
  vpcID:
```

also include the doc updates for real world example for subnets and network ranges